lib/stats.dx

```diff
@@ -0,0 +1,243 @@
+'# Stats
+Probability distributions and other functions useful for statistical computing
+
+
+'## Univariate probability distributions
+Simulation and evaluation of [probability distributions](https://en.wikipedia.org/wiki/Probability_distribution). 
+
+'## Dist interface
+Distributions implement the `Dist` interface.
+
+interface Dist d a
+  log_pdf : d -> a -> Float     -- either the density function or mass function
+  log_cdf : d -> a -> Float     -- the cumulative distribution function (CDF)
+  log_sf : d -> a -> Float      -- the survivor function (complement of CDF)
+  quantile : d -> Float -> a    -- the quantile function (inverse CDF)
+  draw : d -> Key -> a          -- function for random draws
```

Comment:
Once @alexreinking 's user-extensible effects system lands, we'll be able to define a `Random` effect. Then the type of `draw` could be something like `draw : d -> {Random} a`. In that case might get swappable PRNGs automatically, because different handlers for `Random` can use different keys.